src/util/plogic.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+/**
+ * Utilities for mining pool logic, 
+ * this is temporary until we move to the appropriate routes
+ * minerSubmitsBlock() should go into the miner middleware once ready
+ */
+
+moneroApi = require('src/monero.api');
+const cache = require('src/util/cache');
+const cryptonote = require('src/util/cryptonote');
+/**
+ * @Note
+ * @Documentation
+ * @description
+ * We will update the miner profile in redis with the appropriate amount of shares
+ * For now, a cron job should be running to periodically submit shares to sapphire
+ * Ideally we should at first separate traditional and fortune mining to minimize code complexity (traditional requires establishing trust, ban system, etc)
+ * In terms of share grant policy, as we are creating a system first and foremost for fortune mining, we will siply weigh shares by time until a more robust solution is created
+ * These comments are to be moved to proper documentation (TBD)
+ * 
+ * blockTemplate should be provided
+ */
+function minerSubmitsBlock(miner, job, blockTemplate, params){
+
+          // From nodejs pool
+          let nonce = params.nonce;
+          let resultHash = params.result;
+          let template = Buffer.alloc(blockTemplate.buffer.length);
+          
+          blockTemplate.buffer.copy(template);
```

Comment:
There exists either Block.model.js or Template.model.js, it will be nice to refacto blockTemplate here into those files and then instantiate it here and use it here